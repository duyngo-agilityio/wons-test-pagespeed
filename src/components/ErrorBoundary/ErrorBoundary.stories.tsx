import type { Meta, StoryObj } from '@storybook/react';

// Components
import ErrorBoundary from '.';

const meta: Meta<typeof ErrorBoundary> = {
  title: 'Components/ErrorBoundary',
  component: ErrorBoundary,
  parameters: {
    layout: 'centered',
    controls: { expanded: true },
  },
};

export default meta;
type Story = StoryObj<typeof ErrorBoundary>;

export const Default: Story = {
  args: {},
};
