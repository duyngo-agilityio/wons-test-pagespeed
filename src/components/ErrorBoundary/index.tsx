'use client';

import { cn } from '@nextui-org/react';

// Components
import { Button, Heading, Text } from '@/components';

interface ErrorProps {
  error: (Error & { digest?: string }) | string;
  reset?: () => void;
  className?: string;
}

const ErrorBoundary = ({ error, reset, className }: ErrorProps) => {
  // Attempt to recover by trying to re-render the segment
  const handleReset = () => {
    reset?.();
  };

  return (
    <div
      className={cn(
        'text-center h-screen flex justify-center items-center flex-col',
        className,
      )}
    >
      <Heading title="Something went wrong!" className="text-4xl" />
      <Text text={typeof error === 'string' ? error : ''} />
      <Button onClick={handleReset} color="primary" className="mt-8">
        Try again
      </Button>
    </div>
  );
};

export default ErrorBoundary;
