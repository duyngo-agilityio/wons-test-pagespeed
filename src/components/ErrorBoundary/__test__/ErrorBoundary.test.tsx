// Components
import ErrorBoundary from '../index';

const mockProps = {
  error: new Error('Test Error'),
  reset: jest.fn(),
};

describe('ErrorBoundary Component', () => {
  it('Should render ErrorBoundary Component correctly', () => {
    const component = testLibJestUtils.render(<ErrorBoundary {...mockProps} />);

    expect(component).toMatchSnapshot();
  });

  it('renders error message and calls reset function when  button is clicked', () => {
    const { getByText } = testLibJestUtils.render(
      <ErrorBoundary {...mockProps} />,
    );

    const tryAgainButton = getByText('Try again');
    testLibJestUtils.fireEvent.click(tryAgainButton);

    expect(mockProps.reset).toHaveBeenCalled();
  });
});
