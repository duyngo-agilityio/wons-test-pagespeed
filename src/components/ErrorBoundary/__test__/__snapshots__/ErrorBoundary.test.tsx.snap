// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component Should render ErrorBoundary Component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-overlay-container="true"
      >
        <script>
          !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
        </script>
         
        <div
          class="text-center h-screen flex justify-center items-center flex-col"
        >
          <h2
            aria-level="2"
            class="font-bold font-dm-sans text-blue-800 dark:text-white text-5xl text-4xl"
          >
            Something went wrong!
          </h2>
          <button
            class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover bg-blue-500 border-transparent dark:bg-purple-600 text-white hover:bg-blue-100 px-5 pb-2.5 pt-[11px] text-md rounded-[10px] border-[1px] h-auto min-w-max mt-8"
            type="button"
          >
            <div
              class="flex items-center"
            >
              Try again
            </div>
          </button>
        </div>
        <div
          class="fixed top-4 right-4"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-overlay-container="true"
    >
      <script>
        !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
      </script>
       
      <div
        class="text-center h-screen flex justify-center items-center flex-col"
      >
        <h2
          aria-level="2"
          class="font-bold font-dm-sans text-blue-800 dark:text-white text-5xl text-4xl"
        >
          Something went wrong!
        </h2>
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover bg-blue-500 border-transparent dark:bg-purple-600 text-white hover:bg-blue-100 px-5 pb-2.5 pt-[11px] text-md rounded-[10px] border-[1px] h-auto min-w-max mt-8"
          type="button"
        >
          <div
            class="flex items-center"
          >
            Try again
          </div>
        </button>
      </div>
      <div
        class="fixed top-4 right-4"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
